src/neo/SmartContract/ApplicationEngine.OpCodePrices.cs

```diff
@@ -7,190 +7,190 @@ partial class ApplicationEngine
-            [OpCode.JMPNE] = 4,
-            [OpCode.JMPNE_L] = 4,
-            [OpCode.JMPGT] = 4,
-            [OpCode.JMPGT_L] = 4,
-            [OpCode.JMPGE] = 4,
-            [OpCode.JMPGE_L] = 4,
-            [OpCode.JMPLT] = 4,
-            [OpCode.JMPLT_L] = 4,
-            [OpCode.JMPLE] = 4,
-            [OpCode.JMPLE_L] = 4,
-            [OpCode.CALL] = 8,
-            [OpCode.CALL_L] = 8,
-            [OpCode.CALLA] = 16,
+            [OpCode.PUSHINT8] = 1 << 0,
+            [OpCode.PUSHINT16] = 1 << 0,
+            [OpCode.PUSHINT32] = 1 << 0,
+            [OpCode.PUSHINT64] = 1 << 0,
+            [OpCode.PUSHINT128] = 1 << 2,
+            [OpCode.PUSHINT256] = 1 << 2,
+            [OpCode.PUSHA] = 1 << 2,
+            [OpCode.PUSHNULL] = 1 << 0,
+            [OpCode.PUSHDATA1] = 1 << 3,
+            [OpCode.PUSHDATA2] = 1 << 9,
+            [OpCode.PUSHDATA4] = 1 << 12,
+            [OpCode.PUSHM1] = 1 << 0,
+            [OpCode.PUSH0] = 1 << 0,
+            [OpCode.PUSH1] = 1 << 0,
+            [OpCode.PUSH2] = 1 << 0,
+            [OpCode.PUSH3] = 1 << 0,
+            [OpCode.PUSH4] = 1 << 0,
+            [OpCode.PUSH5] = 1 << 0,
+            [OpCode.PUSH6] = 1 << 0,
+            [OpCode.PUSH7] = 1 << 0,
+            [OpCode.PUSH8] = 1 << 0,
+            [OpCode.PUSH9] = 1 << 0,
+            [OpCode.PUSH10] = 1 << 0,
+            [OpCode.PUSH11] = 1 << 0,
+            [OpCode.PUSH12] = 1 << 0,
+            [OpCode.PUSH13] = 1 << 0,
+            [OpCode.PUSH14] = 1 << 0,
+            [OpCode.PUSH15] = 1 << 0,
+            [OpCode.PUSH16] = 1 << 0,
+            [OpCode.NOP] = 1 << 0,
+            [OpCode.JMP] = 1 << 1,
+            [OpCode.JMP_L] = 1 << 1,
+            [OpCode.JMPIF] = 1 << 1,
+            [OpCode.JMPIF_L] = 1 << 1,
+            [OpCode.JMPIFNOT] = 1 << 1,
+            [OpCode.JMPIFNOT_L] = 1 << 1,
+            [OpCode.JMPEQ] = 1 << 1,
+            [OpCode.JMPEQ_L] = 1 << 1,
+            [OpCode.JMPNE] = 1 << 1,
+            [OpCode.JMPNE_L] = 1 << 1,
+            [OpCode.JMPGT] = 1 << 1,
+            [OpCode.JMPGT_L] = 1 << 1,
+            [OpCode.JMPGE] = 1 << 1,
+            [OpCode.JMPGE_L] = 1 << 1,
+            [OpCode.JMPLT] = 1 << 1,
+            [OpCode.JMPLT_L] = 1 << 1,
+            [OpCode.JMPLE] = 1 << 1,
+            [OpCode.JMPLE_L] = 1 << 1,
+            [OpCode.CALL] = 1 << 9,
+            [OpCode.CALL_L] = 1 << 9,
+            [OpCode.CALLA] = 1 << 9,
             [OpCode.ABORT] = 0,
-            [OpCode.ASSERT] = 2,
-            [OpCode.THROW] = 128,
-            [OpCode.TRY] = 4,
-            [OpCode.TRY_L] = 4,
-            [OpCode.ENDTRY] = 4,
-            [OpCode.ENDTRY_L] = 4,
-            [OpCode.ENDFINALLY] = 128,
+            [OpCode.ASSERT] = 1 << 0,
+            [OpCode.THROW] = 1 << 9,
+            [OpCode.TRY] = 1 << 2,
+            [OpCode.TRY_L] = 1 << 2,
+            [OpCode.ENDTRY] = 1 << 2,
+            [OpCode.ENDTRY_L] = 1 << 2,
+            [OpCode.ENDFINALLY] = 1 << 2,
```

Comment:
It can have the same execution time penalty as THROW.

Reply:
But you have paid for `THROW`, right?

Reply:
Yes, you're right about this one, it can only happen after `THROW` and we can assume that it's been paid for.